Invert spotlight angle attenuation

JFonS opened the pull request:
Inverted the spotlight angle attenuation so a higher value results in a dimmer light, this makes it more consistent with the distance attenuation.

Also changed the way spotlighs are computed in SDFGI and GIPorbes, now it matches the falloff used in the scene rendering
code.

This breaks compatibility with 3.2 spotlights, but I think it's worth the change. If we decide against breaking compat, we still need to change SDFGI and GIProbe to use the correct angle attenuation.

fire commented:
It's ok to change this, but a reminder to change the fitted values for https://github.com/godotengine/godot/pull/39450.

lawnjelly commented on servers/rendering/rasterizer_rd/shaders/giprobe.glsl:
```diff
@@ -51,7 +51,7 @@ struct Light {
 
 	vec3 color;
-	float spot_angle_radians;
+	float spot_angle;
```
If these are cos spot angle, would they be better named cos_spot_angle? Just a thought if it made reading easier later on when someone comes back to this.

JFonS commented:
I updated the PR, it should be ready for review and merge.

clayjohn commented:
Are there any changes needed in scene_forward.glsl and volumetric_fog.glsl? I think they probably need to be updated as well.

JFonS commented:
@clayjohn Not really, these are inverted simply because the uniform value is set inverted from the CPU side. The changes in the other places are just to make them match the attenuation in scene_forward and volumetric_fog, they were using a different attenuation formula before and the results were wrong.

clayjohn reviewed (approved):
@JFonS thanks for the clarification. Looks good to me!

akien-mga commented:
Needs rebase (if now is a good time to merge this wrt to the work @reduz is doing on rendering).

JFonS committed:
Invert spotlight angle attenuation

Inverted the spotlight angle attenuation so a higher value results in
a dimmer light, this makes it more consistent with the distance
attenuation.

Also changed the way spotlighs are computed in SDFGI
and GIPorbes and GPU lightmapper, now it matches the falloff used in the scene rendering
code.

JFonS commented:
Rebased, should be good to merge after CI passes.

akien-mga commented:
Thanks!
